React/NextJs/ProjectStructure.md

```diff
@@ -0,0 +1,224 @@
+# Project file structure
+
+## Organizing components
+### UI Components
+
+Before implementing [atomic design methodology](https://github.com/danilowoz/react-atomic-design), our common practice was separating components by the scope of the page + core components + shared components like this:
+
+Lowercase folders were indicators of a page scope (except of core).
+
+```
+.
+├── src
+└── components
+    ├── core
+    │   └── button
+    │       ├── Button.tsx
+    │       └── stories.tsx
+    ├── admin
+    │   └── welcome-banner
+    │       └── WelcomeBanner.tsx
+    ├── carousel
+    │   └── Carousel.tsx
+    └── card
+        └── Card.tsx
+```
+
+With atomic design, we would completely separate all UI components and build them as a blocks of atoms, molecules and organisms (together with templates and pages for putting everything together at the end) 
+
+```
+src
+.
+└── components
+    └── ui
+        ├── atoms
+        │   └── button
+        │       ├── Button.tsx
+        │       └── stories.tsx
+        ├── molecules
+        │   ├── card
+        │   │   └── Card.tsx
+        │   └── welcome-banner
+        │       └── WelcomeBanner.tsx
+        └── organisms
+            └── carousel
+                └── Carousel.tsx
+```
+
+## Different molecule layouts
+
+Sometimes, you might want to create a molecule specific for mobile and desktop. You might use a tool like [fresnel](https://github.com/artsy/fresnel) for detecting mediaqueries.
+
+In this case, inside a specific molecule, we could add a subfolder `layouts` (not to be confused with actual layout described below) to define how our molecule would look like on specific media query.
+
+```
+.
+└── molecules
+    └── card
+        ├── layouts
+        │   ├── Desktop.tsx
```

Comment:
Can we name it like this
`Card.desktop.tsx`, `Card.mobile.tsx`, `Card.tablet.tsx`
it is easier to search for a component.
And react by itself will introduce naming like this with server-side components, e.g. `Card.server.tsx` and `Card.client.tsx`